CMakeLists.txt

```diff
@@ -134,28 +137,84 @@ endif()
 
 if (NOT "${Boost_VERSION}" STRLESS "106100")
     find_package(OpenMP)
-    add_library(JIT INTERFACE)
-    add_library(VexCL::JIT ALIAS JIT)
-
-    # CMake 3.9 FindOpenMP allows correct linking with Clang in more cases
-    if(TARGET OpenMP::OpenMP_CXX)
-        target_link_libraries(JIT INTERFACE OpenMP::OpenMP_CXX Common)
-    else()
-        # Clang may need -fopenmp=libiomp5 instead, can't be detected here without CMake 3.9
-        target_link_libraries(JIT INTERFACE
-            $<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
-            $<$<CXX_COMPILER_ID:Clang>:${OpenMP_CXX_FLAGS}>
-            $<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
-            )
-        target_compile_options(JIT INTERFACE ${OpenMP_CXX_FLAGS})
-    endif()
 
-    target_link_libraries(JIT INTERFACE Common ${CMAKE_DL_LIBS})
-    target_compile_definitions(JIT INTERFACE VEXCL_BACKEND_JIT)
+    # Have to check OpenMP_CXX_FOUND due to bug in
+    # one version of CMake (fixed in patch release)
+    # OpenMP is missing on macOS llvm default, for example
+    if(OpenMP_FOUND OR OpenMP_CXX_FOUND)
+
+        add_library(JIT INTERFACE)
+        add_library(VexCL::JIT ALIAS JIT)
+
+        # CMake 3.9 FindOpenMP allows correct linking with Clang in more cases
+        if(TARGET OpenMP::OpenMP_CXX)
+            target_link_libraries(JIT INTERFACE OpenMP::OpenMP_CXX Common)
+        else()
+            # Clang may need -fopenmp=libiomp5 instead, can't be detected here without CMake 3.9
+            target_link_libraries(JIT INTERFACE
+                $<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
+                $<$<CXX_COMPILER_ID:Clang>:${OpenMP_CXX_FLAGS}>
+                $<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
+                )
+            target_compile_options(JIT INTERFACE ${OpenMP_CXX_FLAGS})
+        endif()
+
+        target_link_libraries(JIT INTERFACE Common ${CMAKE_DL_LIBS})
+        target_compile_definitions(JIT INTERFACE VEXCL_BACKEND_JIT)
 
-    message(STATUS "Found VexCL::JIT")
```

Comment:
We still report other found backends. Why remove this?

Reply:
Oops, that would be a mistake.

Reply:
I think it happened when I rebased. Fixed.